SingularityService/src/main/java/com/hubspot/singularity/mesos/SingularityMesosSchedulerClient.java

```diff
@@ -358,18 +368,36 @@ public void sendCall(Call call) {
         SinkOperations.create(
           call,
           () -> LOG.debug("Finished {} call to mesos", call.getType()),
-          t -> {
-            LOG.error(
-              "Exception calling mesos ({} so far)",
-              failedMesosCalls.incrementAndGet(),
-              t
-            );
-          }
+          this::checkAndReconnect
         )
       )
     );
   }
 
+  public CompletableFuture<Void> checkAndReconnect(Throwable t) {
+    LOG.error(
+      "Exception calling mesos ({} so far)",
+      failedMesosCalls.incrementAndGet(),
+      t
+    );
+
+    String message = t.getMessage();
+
+    if (
+      message != null &&
+      message.equals(
+        "Error while trying to send request. Status: 403 Message: \'Framework is not subscribed\'"
+      )
+    ) {
```

Comment:
I think we can make this a bit more flexible so we aren't at the mercy of the exception message. I think this is a valid flow for any time we get a `Mesos4xxException` exception with a 403 in the message (or if the exception itself has a status code in it that would be even better to compare to the int vs a string.contains)

Reply:
I can update to check that the message contains 403 but I removed the Mesos4xxException check earlier because I can't mock it because it's a final class